Eastern-Research-Group/csb-rebate-forms-app
# Feature/refactor crf api requests

## Related Issues:
* CSBAPP-352

## Main Changes:
Refactor the 2022 CRF API logic into common functions that will support both the 2022 and 2023 CRF forms (similar to what was done for the FRF and PRF forms). No functional change for the 2022 CRFs – just preparing wrapper app to support eventual 2023 CRFs.

## Steps To Test:
1. Navigate to the dashboard.
2. Ensure at least one 2022 PRF has been selected (BAP "Accepted" status).
3. Click "New Close Out" button – ensure a new close out form is created.
4. Click "Next" on the form, and ensure the form is updated.
5. Return to the user dashboard.
6. Click "Edit" beside the newly created 2023 CRF, and ensure it loads.
7. Click "Next" again on the form (possibly changing form field values as needed), and ensure form is updated.
